server/src/test/java/org/elasticsearch/cluster/routing/allocation/RoutingNodesTests.java

```diff
@@ -484,18 +478,13 @@ public void testMoveShardWithPromotableOnlyRole() {
 
         var routingNodes = clusterState.getRoutingNodes().mutableCopy();
 
-        routingNodes.relocateOrReinitializeShard(
-            routingNodes.node("node-1").getByShardId(shardId),
-            "node-3",
-            0L,
-            new RoutingChangesObserver() {
-            }
-        );
+        routingNodes.relocateShard(routingNodes.node("node-1").getByShardId(shardId), "node-3", 0L, new RoutingChangesObserver() {
+        });
 
-        assertThat(routingNodes.node("node-1").getByShardId(shardId), nullValue());
-        assertThat(routingNodes.node("node-2").getByShardId(shardId), nullValue());
+        assertThat(routingNodes.node("node-1").getByShardId(shardId).state(), equalTo(RELOCATING));
```

Comment:
This is the only part that I had not seen before in your changes. I see we're asserting an intermediate state here where the relocation has not fully completed yet (it's in transit from node-1 to node-3). Can we also wait for the completion of the relocation here and assert that it's finally only in node-3?

Reply:
Yeah we hadn't run these tests before, only the serverless ones, so I expect there might be a few changes needed.

Reply:
There is no actual relocation in progress here, this is only to verify that the routing table changes as we expect. This test (and the one above) were added to support the temporary change that triggered stateless relocations by just closing all the shards, so now really this test can say that the routing table behaves the same for stateless and regular relocations. I've removed the duplication in d4ca54f.